mk/board.mk

```diff
@@ -0,0 +1,3 @@
+ifeq ($(BOARD),de10nano)
+    CFLAGS += -mcpu=cortex-a9 -mtune=cortex-a9 -mfloat-abi=softfp -mfpu=neon
```

Comment:
I have no idea why you sticked on `softfp` instead of `hardfp`. Is it limited by prebuilt root file system?

Reply:
Yes.
If we use `-mfloat-abi=hard`, the compiler says it can not find `gnu/stubs-hard.h` header file and I am sure about that it does not exist on the file system.

Reply:
Can you validate with latest Linaro toolchain?

Reply:
Our **GCC** version is 6.3.0.
The [newest binary release version](https://releases.linaro.org/components/toolchain/binaries/latest-7/arm-linux-gnueabihf/) on the Linaro official website is 7.4.1.
And I do find the file `gnu/stubs-hard.h` in the newest release.
I will download the source code, build and install it.
Or is there any easier way to validate?

Reply:
If Debian GNU/Linux is installed on de10-nano, it is always straightforward to validate latest stable packages.

Reply:
The ARM official website and the Linaro download webpage provide the cross compiler instead of the native compiler.
I'm afraid that I can only validate it with the cross compiler since I found no resource of the native compiler.
And building a native compiler on `DE10 nano` board would be a tedious work.

Reply:
For automated builds, we shall always use cross-compilation.

Reply:
After trying with the hardware floating-point cross compiler, I found out the executables can not be run on `DE10 nano` board since we do not have the corresponding dynamic loader(interpreter).
I think I can mention the `-mfloat-abi=hard` option as a to-do in the comment.

Reply:
Can you check this? https://www.digikey.com/eewiki/display/linuxonarm/DE0-Nano-SoC+Kit
I think we shall not support softfp since it is ABI-incompatible with latest Debian GNU/Linux.

Reply:
I have already followed the steps in the [link](https://www.digikey.com/eewiki/display/linuxonarm/DE0-Nano-SoC+Kit) and even used the newest cross compiler.
In order to validate the `-mfloat-abi=hard`, I think we might need to try another image file rather than the one offered by [LampaLab/iota_fpga](https://github.com/LampaLab/iota_fpga/releases/tag/v0.1)?
We are still lacking of the hardware floating-point related compiler and dynamic loader.

Reply:
Or maybe try to install those compiler and dynamic loader on the `DE10 nano` board.

Reply:
Don't depend on the old image file provided by Lampa Lab since it was indeed out-of-date. We can install Debian-based images on de10-nano and then cross-compile kernel module and dcurl library for usage. hardfp ABI avoids the overhead of moving data between integer and floating point registers but also renders the code incompatible with code built with other -mfloat-abi settings.

Reply:
Performance impact caused by softfp: https://github.com/xiph/opus/issues/93

Reply:
Based on https://github.com/DLTcollab/dcurl/issues/178, the compiler option `-mfloat-abi=hard` works well if we use the armhf based Linux distribution on de10-nano board.